File: lesson5/HW/controller/UserController.java
package lesson5.HW.controller;

import java.util.List;

import lesson5.HW.model.User;
import lesson5.HW.service.UserService;
import lesson5.HW.service.UserServiceImpl;

public class UserController {

	private UserService userService = new UserServiceImpl();

	public void saveUser(String name, String surname, String patronymic) {
		userService.saveUser(new User(name, surname, patronymic));
	}

	public void deleteUser(User user) {
		userService.deleteUser(user);
	}

	public void updateUser(int index, User user) {
		userService.updateUser(index, user);
	}

	public List<User> getAllUsers() {
		return userService.getAllUsers();
	}

	public void addFunds(int index, float funds) {
		userService.addFunds(index, funds);
	}

	public void removeFunds(int index, float funds) {
		userService.removeFunds(index, funds);
	}
}

File: Seminar7/DZ/py/UserController.py
from UserServiceImpl import UserServiceImpl
from User import User


class UserController:

    __userService = UserServiceImpl()

    def __init__(self):
        self.__userService = UserServiceImpl()

    def saveUser(self, name, surname, patronymic):
        self.__userService.saveUser(User(name, surname, patronymic))

    def deleteUser(self, user):
        if isinstance(user, User):
            self.__userService.deleteUser(user)

    def updateUser(self, index, user):
        if isinstance(user, User):
            self.__userService.updateUser(index, user)

    def getAllUsers(self):
        return self.__userService.getAllUsers()

    def addFunds(self, index, funds):
        self.__userService.addFunds(index, funds)

    def removeFunds(self, index, funds):
        self.__userService.removeFunds(index, funds)
